### PR TITLE
feat: GridBinFinder also allows a single pair of ints

### PR DESCRIPTION
**Pull request by CarloVarni**
As of now the `GridBinFinder` only allows as inputs `int` or `std::vector<std::pair<int, int>>` types. With this PR I'm also allowing `std::pair<int, int>`

In this way, the used can define a pair `{k, l}` that will instruct the code to search (within a grid axis) for the `k-` and `l-` neighbouring bins. Thus, given a bin `N`, the GridBinFinder will be instructed to return `N-k, ...., N, ....N+l` bins

### CODE REVIEW

File: Core/include/Acts/Utilities/GridBinFinder.ipp [hunk 1]
```diff
@@ -10,10 +11,12 @@ template <std::size_t DIM>
...
-  static_assert(std::conjunction<std::disjunction<
-                    std::is_same<int, typename std::decay<args>::type>,
-                    std::is_same<std::vector<std::pair<int, int>>,
-                                 typename std::decay<args>::type>>...>::value);
+  static_assert(
+      std::conjunction<std::disjunction<
+          std::is_same<int, typename std::decay<args>::type>,
+          std::is_same<std::pair<int, int>, typename std::decay<args>::type>,
+          std::is_same<std::vector<std::pair<int, int>>,
+                       typename std::decay<args>::type>>...>::value);
```
Comment (paulgessinger) [inline Core/include/Acts/Utilities/GridBinFinder.ipp:19]: Are you reimplementing function / constructor overloading here with variadic template parameters? 😅

Comment (CarloVarni) [reply Core/include/Acts/Utilities/GridBinFinder.ipp:19]: that is just checking the provided inputs are of the allowed types. Do you suggest doing it in a different way?

**Review by paulgessinger — APPROVED** [review]
Seems good as far as I can tell.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [6887847d7db0273cd713c132496663d8572e4955]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374477)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374476)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374475)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374474)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374473)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374472)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374471)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374470)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374469)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374468)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374467)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374466)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374465)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374464)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374463)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374462)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374461)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374460)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374459)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374458)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374457)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374456)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35374455)||

**Pull request merged**